src/main/java/org/skriptlang/skript/lang/comparator/Relation.java

```diff
@@ -0,0 +1,185 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.lang.comparator;
+
+/**
+ * Represents a relation between two objects.
+ */
+public enum Relation {
+
+	EQUAL, NOT_EQUAL, GREATER, GREATER_OR_EQUAL, SMALLER, SMALLER_OR_EQUAL;
+
+	/**
+	 * @param b The boolean to get a Relation from.
+	 * @return {@link #EQUAL} if 'b' is true else {@link #NOT_EQUAL}
+	 */
+	public static Relation get(boolean b) {
+		return b ? Relation.EQUAL : Relation.NOT_EQUAL;
+	}
+
+	/**
+	 * @param i The int to get a Relation from.
+	 * @return {@link #EQUAL} if 'i' is equal to 0,
+	 * {@link #GREATER} if 'i' is greater than 0,
+	 * {@link #SMALLER} if 'i' is less than 0
+	 */
+	public static Relation get(int i) {
+		return i == 0 ? Relation.EQUAL : i > 0 ? Relation.GREATER : Relation.SMALLER;
+	}
+
+	/**
+	 * @param d The double to get a Relation from.
+	 * @return {@link #EQUAL} if 'd' is equal to 0,
+	 * {@link #GREATER} if 'd' is greater than 0,
+	 * {@link #SMALLER} if 'd' is less than 0
+	 */
+	public static Relation get(double d) {
+		return d == 0 ? Relation.EQUAL : d > 0 ? Relation.GREATER : Relation.SMALLER;
+	}
+
+	/**
+	 * Test whether this Relation is fulfilled if another is, i.e. if the parameter 'other' fulfils <code>X rel Y</code>,
+	 * then this Relation fulfils <code>X rel Y</code> as well.
+	 *
+	 * @param other The Relation to compare with.
+	 * @return Whether this Relation is part of the given Relation, e.g. <code>GREATER_OR_EQUAL.is(EQUAL)</code> returns true.
+	 */
+	public boolean is(Relation other) {
```

Comment:
What do you think about naming this method `implies`?

Reply:
Yeah, I think that is a clearer name for the method - it shouldn't be hard for an addon to switch anyways